client/packages/system/src/Name/ListView/ImportProperties/PropertiesImportModal.tsx

```diff
@@ -0,0 +1,239 @@
+  facilities: FacilityNameRowFragment[] | undefined;
+}
+
+export type ImportRow = {
+  code: string;
+  name: string;
+  properties: Record<string, string>;
+  id: string;
+  errorMessage?: string;
+};
+
+enum Tabs {
+  Upload = 'Upload',
+  Review = 'Review',
+  Import = 'Import',
+}
+
+export type LineNumber = {
+  lineNumber: number;
+};
+
+const toUpdateNamePropertiesInput = (
+  row: ImportRow
+): UpdateNamePropertiesInput => {
+  return { id: row.id, properties: JSON.stringify(row.properties) };
+};
+
+export const PropertiesImportModal: FC<PropertiesImportModalProps> = ({
+  isOpen,
+  onClose,
+  facilities,
+}) => {
+  const t = useTranslation();
+  const { success } = useNotification();
+  const { currentTab, onChangeTab } = useTabs(Tabs.Upload);
+  const [activeStep, setActiveStep] = useState(0);
+  const { Modal } = useDialog({ isOpen, onClose });
+
+  const [errorMessage, setErrorMessage] = useState<string>(() => '');
+  const [importProgress, setImportProgress] = useState(0);
+  const [importErrorCount, setImportErrorCount] = useState(0);
+
+  const [warningMessage] = useState<string>(() => '');
+  const { data: properties } = useNameProperties();
+  const { mutateAsync } = useName.document.updateProperties('');
+
+  const [bufferedFacilityProperties, setBufferedFacilityProperties] = useState<
+    ImportRow[]
+  >(() => []);
+
+  const importAction = async () => {
+    onChangeTab(Tabs.Import);
+    const numberImportRecords = bufferedFacilityProperties?.length ?? 0;
+    if (bufferedFacilityProperties && numberImportRecords > 0) {
+      const importErrorRows: ImportRow[] = [];
+      // Import count can be quite large, we break this into blocks of 10 to avoid too much concurrency
+      const remainingRecords = bufferedFacilityProperties;
+      while (remainingRecords.length) {
+        await Promise.all(
+          remainingRecords.splice(0, 10).map(async facility => {
+            await mutateAsync(toUpdateNamePropertiesInput(facility))
+              .then(async result => {
+                // Map structured Errors
+                if (result?.__typename === 'Mutations') {
+                  // const errorMessage = mapStructuredErrors(result);
+                  const errorMessage = result;
+                  importErrorRows.push({
+                    ...facility,
+                    errorMessage: errorMessage.__typename,
+                  });
+                  return;
+                }
+              })
+              .catch((err: { message: string }) => {
+                if (!err) {
+                  err = { message: t('messages.unknown-error') };
+                }
+                importErrorRows.push({
+                  ...facility,
+                  errorMessage: err.message,
+                });
+              });
+          })
+        ).then(() => {
+          // Update Progress Bar
+          const percentComplete =
+            10 - (remainingRecords.length / numberImportRecords) * 100.0;
```

Comment:
```suggestion
            100 - (remainingRecords.length / numberImportRecords) * 100.0;
```

To make the percentage complete?